src/Collection.php

```diff
@@ -203,18 +203,25 @@ public function aggregate(array $pipeline, array $options = [])
 
         /* A "majority" read concern is not compatible with the $out stage, so
          * avoid providing the Collection's read concern if it would conflict.
+         *
+         * A read concern is also not compatible with transactions.
          */
         if ( ! isset($options['readConcern']) &&
-             ! ($hasOutStage && $this->readConcern->getLevel() === ReadConcern::MAJORITY) &&
-            \MongoDB\server_supports_feature($server, self::$wireVersionForReadConcern)) {
+            ! ($hasOutStage && $this->readConcern->getLevel() === ReadConcern::MAJORITY) &&
+            \MongoDB\server_supports_feature($server, self::$wireVersionForReadConcern) &&
+            ! $this->isInTransaction($options)) {
             $options['readConcern'] = $this->readConcern;
         }
 
         if ( ! isset($options['typeMap'])) {
             $options['typeMap'] = $this->typeMap;
         }
 
-        if ($hasOutStage && ! isset($options['writeConcern']) && \MongoDB\server_supports_feature($server, self::$wireVersionForWritableCommandWriteConcern)) {
+        if ($hasOutStage &&
+            ! isset($options['writeConcern']) &&
+            \MongoDB\server_supports_feature($server,
+                self::$wireVersionForWritableCommandWriteConcern) &&
```

Comment:
Does this function call need to be split across two lines? It should fall within 120 characters easily.